index.js

```diff
@@ -69,10 +71,11 @@ exports.handler = function(event, context) {
 					}
 				});
 			}, function(err) {
+				context.callbackWaitsForEmptyEventLoop = false;
 				if (err) {
-					context.fail(err);
+					callback(err, 'Failed result');
```

Comment:
usually, if the first parameter is set (an error), the second one is not passed.